### PR TITLE
Fix for statusmonitor that fails on control component

### PR DESCRIPTION
**Pull request by Dysproz**
This review fixes error in control components which fails with various errors like: `json: cannot unmarshal string into Go struct field .BgpRouterState.num_up_xmpp_peer of type uves.Num`

Errors appear because payload returned to statusmonitor looks like that:
```
...
        "num_up_xmpp_peer": [
            [
                {
                    "@type": "u32",
                    "#text": "0"
                },
                "ip-10-0-142-183.eu-central-1.compute.internal:Control:contrail-control:0"
            ]
        ],
...
```

where list contains one dict and one string. Current logic tries to unmarshall this string to two-field struct.

In this review I fix that problem with custom type which allows to gather status (because for status we use only `#text`  field value.

### CODE REVIEW

**Comment by Dysproz** [comment]
recheck

**Comment by Dysproz** [comment]
recheck

File: statusmonitor/uves/control.go [hunk 1]
```diff
@@ -75,38 +108,38 @@ type ControlUVEStatus struct {
 	} `json:"NodeStatus"`
 
 	BgpRouterState struct {
-		NumDownServiceChains      [][]interface{}         `json:"num_down_service_chains"`
-		BgpRouterIPList           [][]interface{}         `json:"bgp_router_ip_list"`
-		NumUpXMPPPeer             [][]NumUpXMPPPeer       `json:"num_up_xmpp_peer"`
```
Comment (iosetek) [inline statusmonitor/uves/control.go:80]: If structure NumUpXMPPPeer is not used anymore please remove it.

File: statusmonitor/uves/control_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+					},
+				},
+				{
+					map[string]string{
+						"@type": "u32",
+						"#text": "6",
+					},
+				},
+			},
+			want:    "",
```
Comment (iosetek) [inline statusmonitor/uves/control_test.go:74]: Make the zero value useful.
This line is redundant.

File: statusmonitor/uves/control_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+			s: [][]interface{}{
+				{
+					map[string]string{
+						"@type": "u32",
+						"#text": "5",
+					},
+				},
+			},
+			want:    "5",
+			wantErr: false,
```
Comment (iosetek) [inline statusmonitor/uves/control_test.go]: Make the zero value useful.
This line is redundant.

File: statusmonitor/main.go [hunk 4]
```diff
@@ -415,48 +415,53 @@ func getControlStatusFromResponse(statusBody []byte) *contrailOperatorTypes.Cont
...
 	staticRoutes := contrailOperatorTypes.StaticRoutes{}
 	bgpPeer := contrailOperatorTypes.BGPPeer{}
 	state := "down"
 
 	if controlUVEStatus != nil {
-		if len(controlUVEStatus.BgpRouterState.NumDownStaticRoutes) > 0 && len(controlUVEStatus.BgpRouterState.NumDownStaticRoutes[0]) > 0 && controlUVEStatus.BgpRouterState.NumDownStaticRoutes[0][0].Text != "" {
-			numDownStaticRoutes = controlUVEStatus.BgpRouterState.NumDownStaticRoutes[0][0].Text
+		numDownStaticRoutes, err := controlUVEStatus.BgpRouterState.NumDownStaticRoutes.Status()
+		if err != nil {
+			log.Fatalf("numDownStaticRoutes", err)
```
Comment (iosetek) [inline statusmonitor/main.go]: You changed the code logic by adding Fatal here. Please change it to Info/Debug

File: statusmonitor/uves/control.go [hunk 5]
```diff
@@ -1,9 +1,42 @@
...
+	statusKey = "#text"
+)
+
+type statusData [][]interface{}
+
+func (s statusData) Status() (string, error) {
+	if len(s) > 1 {
+		return "", errors.New("Status payload has more fields than expected")
+	} else if len(s) == 0 {
+		return "0", nil
```
Comment (iosetek) [inline statusmonitor/uves/control.go]: Add constant for "0" as defaultValue.

File: statusmonitor/uves/control.go (hunk 1 above)
Comment (Dysproz) [reply statusmonitor/uves/control.go:80]: Done

File: statusmonitor/uves/control_test.go (hunk 2 above)
Comment (Dysproz) [reply statusmonitor/uves/control_test.go]: Done, because I removed errors

File: statusmonitor/uves/control_test.go (hunk 3 above)
Comment (Dysproz) [reply statusmonitor/uves/control_test.go]: no more errors :)

File: statusmonitor/main.go (hunk 4 above)
Comment (Dysproz) [reply statusmonitor/main.go]: Removed errors as there were none

File: statusmonitor/uves/control.go (hunk 5 above)
Comment (Dysproz) [reply statusmonitor/uves/control.go]: Done

**Pull request merged**